miden/Cargo.toml

```diff
@@ -31,6 +31,7 @@ hex = { version = "0.4", optional = true }
 log = { version = "0.4", default-features = false }
 verifier = { package = "miden-verifier", path = "../verifier", version = "0.2", default-features = false }
 vm-core = { package = "miden-core", path = "../core", version = "0.2", default-features = false }
+rand = { version = "0.8.4", optional = false }
```

Comment:
This should probably be a dev dependency. Also, I wonder if we could just rely on `rand_utils` - e.g., by generating `u16` values and then modding them by the upper bound or something like that.

Reply:
@bobbinth I removed the random value tests and moved them to proptests so rand package was no longer required. Please let me know if random value tests are still required. Seemed redundant after adding proptests.

Reply:
@tohrnii yep, I agree with you. No need to do the random value tests - using proptests the way you've done is great.